discord_bots/cogs/in_progress_game.py

```diff
@@ -111,151 +139,281 @@ def get_player_and_in_progress_game(
+        outcome: Literal["win", "loss", "tie"],
+        game_id: Optional[str] = None,
     ) -> bool:
         if not interaction.response.is_done():
             await interaction.response.defer(ephemeral=True)
         if _lock.locked():
             await interaction.followup.send(
-                embed=discord.Embed(
-                    description="⏳ A **Cancel** is already in progress, please wait...",
-                    color=discord.Color.yellow(),
+                embed=Embed(
+                    description="⏳ A **Finish** is already in progress, please wait...",
+                    color=Colour.yellow(),
                 ),
                 ephemeral=True,
             )
         async with _lock:
-            session: sqlalchemy.orm.Session
+            session: SQLAlchemySession
             with Session() as session:
-                game = (
-                    session.query(InProgressGame)
-                    .filter(InProgressGame.id.startswith(game_id))
-                    .first()
+                result = self.get_player_and_in_progress_game(
+                    session, interaction.user.id, game_id
                 )
-                if not game:
+                if result is None:
+                    embed = Embed(
+                        color=Colour.red(),
+                    )
+                    embed.description = "❌ You are not in this game"
                     await interaction.followup.send(
-                        embed=discord.Embed(
-                            description=f"❌ Game {short_uuid(game_id)} does not exist",
-                            colour=discord.Colour.red(),
-                        ),
+                        embed=embed,
                         ephemeral=True,
                     )
                     return False
+
+                game_player: InProgressGamePlayer = result[0]
+                game: InProgressGame = result[1]
                 confirmation_buttons = ConfirmationView(interaction.user.id)
                 confirmation_buttons.message = await interaction.followup.send(
-                    embed=discord.Embed(
-                        description=f"⚠️ Are you sure you want to **Cancel** game **{short_uuid(game.id)}**?⚠️",
-                        color=discord.Colour.yellow(),
+                    embed=Embed(
+                        description=f"⚠️ Are you sure you want to **Finish** game **{short_uuid(game.id)}** as a **{outcome}**?⚠️",
+                        color=Colour.yellow(),
                     ),
                     view=confirmation_buttons,
                     ephemeral=True,
                 )
                 await confirmation_buttons.wait()
                 if not confirmation_buttons.value:
                     return False
-                is_game_finished = await self.cancel_in_progress_game(
-                    session, interaction, game
+                is_finished = await self.finish_in_progress_game(
```

Comment:
Sorry, I got confused by the diff and thought you changed this above. Rename this to `is_game_finished`

Reply:
Renamed